### ISSUE TITLE
License class 'reachable: false' not collecting MAC address properly from device-info

### ISSUE THREAD

**Issue by jgruber** [F5Networks/f5-declarative-onboarding#117]
### Environment
 * Declarative Onboarding Version:
 * BIG-IP Version:

### Summary
Please also include information about the reproducibility and the severity/impact of the issue.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
   "schemaVersion" : "1.0.0",
   "async" : true,
   "Common" : {
      "selfip_tapdfcfb2fa_12" : {
         "address" : "192.168.40.43/24",
         "class" : "SelfIp",
         "trafficGroup" : "traffic-group-local-only",
         "allowService" : "all",
         "vlan" : "net_tapdfcfb2fa_12"
      },
      "configsync" : {
         "configsyncIp" : "1.1.1.29",
         "class" : "ConfigSync"
      },
      "trust" : {
         "remoteHost" : "1.1.1.29",
         "localUsername" : "admin",
         "localPassword" : "f5c0nfig",
         "class" : "DeviceTrust",
         "remoteUsername" : "admin",
         "remotePassword" : "f5c0nfig"
      },
      "activeStandbyGroup" : {
         "owner" : "waf4nic-secondary.openstack.grubernet.org",
         "saveOnAutoSync" : false,
         "networkFailover" : true,
         "fullLoadOnSync" : false,
         "autoSync" : false,
         "asmSync" : false,
         "members" : [
            "waf4nic-primary.openstack.grubernet.org",
            "waf4nic-secondary.openstack.grubernet.org"
         ],
         "class" : "DeviceGroup",
         "type" : "sync-failover"
      },
      "hostname" : "waf4nic-secondary.openstack.grubernet.org",
      "tap7c3c7447-f9_default_gw" : {
         "network" : "default",
         "gw" : "192.168.80.1",
         "class" : "Route",
         "mtu" : 0
      },
      "selfip_tap661eb5ae_89" : {
         "address" : "1.1.1.29/24",
         "class" : "SelfIp",
         "trafficGroup" : "traffic-group-local-only",
         "allowService" : "all",
         "vlan" : "net_tap661eb5ae_89"
      },
      "dbvars" : {
         "ui.advisory.enabled" : true,
         "ui.advisory.color" : "orange",
         "class" : "DbVariables",
         "ui.advisory.text" : "This device is under centralized management."
      },
      "dnsServers" : {
         "search" : [
            "openstack.grubernet.org"
         ],
         "class" : "DNS",
         "nameServers" : [
            "8.8.8.8"
         ]
      },
      "provisioningLevels" : {
         "ltm" : "nominal",
         "class" : "Provision",
         "asm" : "nominal"
      },
      "net_tap7c3c7447_f9" : {
         "interfaces" : [
            {
               "tagged" : false,
               "name" : "1.3"
            }
         ],
         "class" : "VLAN",
         "tag" : 4091,
         "mtu" : 1450
      },
      "poolLicense" : {
         "reachable" : false,
         "licensePool" : "BIGIPVEREGKEYS",
         "hypervisor" : "kvm",
         "bigIqHost" : "172.13.1.108",
         "bigIqPassword" : "admin",
         "licenseType" : "licensePool",
         "bigIqUsername" : "admin",
         "class" : "License"
      },
      "failoverAddress" : {
         "address" : "1.1.1.29",
         "class" : "FailoverUnicast"
      },
      "ntpServers" : {
         "timezone" : "UTC",
         "servers" : [
            "0.pool.ntp.org",
            "1.pool.ntp.org"
         ],
         "class" : "NTP"
      },
      "selfip_tap7c3c7447_f9" : {
         "address" : "192.168.80.40/24",
         "class" : "SelfIp",
         "trafficGroup" : "traffic-group-local-only",
         "allowService" : "all",
         "vlan" : "net_tap7c3c7447_f9"
      },
      "class" : "Tenant",
      "net_tapdfcfb2fa_12" : {
         "interfaces" : [
            {
               "tagged" : false,
               "name" : "1.2"
            }
         ],
         "class" : "VLAN",
         "tag" : 4092,
         "mtu" : 1450
      },
      "net_tap661eb5ae_89" : {
         "interfaces" : [
            {
               "tagged" : false,
               "name" : "1.1"
            }
         ],
         "class" : "VLAN",
         "tag" : 4093,
         "mtu" : 1450
      }
   },
   "label" : "generated from OpenStack metadata",
   "class" : "Device"
}

```

2. Observe the following error response:
```json
{
   "errors" : [
      "macAddress is required when assignmentType is UNREACHABLE",
      "macAddress is required when assignmentType is UNREACHABLE"
   ],
   "status" : "ERROR",
   "message" : "invalid config - rolled back",
   "selfLink" : "https://localhost/mgmt/shared/declarative-onboarding/task/d3ab0e60-b9da-46e1-9ccd-a9782cda1481",
   "declaration" : {
      "schemaVersion" : "1.0.0",
      "async" : true,
      "Common" : {
         "selfip_tapdfcfb2fa_12" : {
            "address" : "192.168.40.43/24",
            "class" : "SelfIp",
            "trafficGroup" : "traffic-group-local-only",
            "allowService" : "all",
            "vlan" : "net_tapdfcfb2fa_12"
         },
         "configsync" : {
            "configsyncIp" : "1.1.1.29",
            "class" : "ConfigSync"
         },
         "trust" : {
            "remoteHost" : "1.1.1.29",
            "localUsername" : "admin",
            "class" : "DeviceTrust",
            "remoteUsername" : "admin"
         },
         "activeStandbyGroup" : {
            "owner" : "waf4nic-secondary.openstack.grubernet.org",
            "saveOnAutoSync" : false,
            "networkFailover" : true,
            "fullLoadOnSync" : false,
            "autoSync" : false,
            "asmSync" : false,
            "members" : [
               "waf4nic-primary.openstack.grubernet.org",
               "waf4nic-secondary.openstack.grubernet.org"
            ],
            "class" : "DeviceGroup",
            "type" : "sync-failover"
         },
         "hostname" : "waf4nic-secondary.openstack.grubernet.org",
         "tap7c3c7447-f9_default_gw" : {
            "network" : "default",
            "gw" : "192.168.80.1",
            "class" : "Route",
            "mtu" : 0
         },
         "selfip_tap661eb5ae_89" : {
            "address" : "1.1.1.29/24",
            "class" : "SelfIp",
            "trafficGroup" : "traffic-group-local-only",
            "allowService" : "all",
            "vlan" : "net_tap661eb5ae_89"
         },
         "dbvars" : {
            "ui.advisory.enabled" : true,
            "ui.advisory.color" : "orange",
            "class" : "DbVariables",
            "ui.advisory.text" : "This device is under centralized management."
         },
         "dnsServers" : {
            "search" : [
               "openstack.grubernet.org"
            ],
            "class" : "DNS",
            "nameServers" : [
               "8.8.8.8"
            ]
         },
         "provisioningLevels" : {
            "ltm" : "nominal",
            "class" : "Provision",
            "asm" : "nominal"
         },
         "net_tap7c3c7447_f9" : {
            "interfaces" : [
               {
                  "tagged" : false,
                  "name" : "1.3"
               }
            ],
            "cmpHash" : "default",
            "class" : "VLAN",
            "tag" : 4091,
            "mtu" : 1450
         },
         "poolLicense" : {
            "reachable" : false,
            "licensePool" : "BIGIPVEREGKEYS",
            "unitOfMeasure" : "monthly",
            "hypervisor" : "kvm",
            "bigIqHost" : "172.13.1.108",
            "licenseType" : "licensePool",
            "bigIqUsername" : "admin",
            "overwrite" : false,
            "class" : "License"
         },
         "failoverAddress" : {
            "address" : "1.1.1.29",
            "class" : "FailoverUnicast",
            "port" : 1026
         },
         "ntpServers" : {
            "timezone" : "UTC",
            "servers" : [
               "0.pool.ntp.org",
               "1.pool.ntp.org"
            ],
            "class" : "NTP"
         },
         "selfip_tap7c3c7447_f9" : {
            "address" : "192.168.80.40/24",
            "class" : "SelfIp",
            "trafficGroup" : "traffic-group-local-only",
            "allowService" : "all",
            "vlan" : "net_tap7c3c7447_f9"
         },
         "class" : "Tenant",
         "net_tapdfcfb2fa_12" : {
            "interfaces" : [
               {
                  "tagged" : false,
                  "name" : "1.2"
               }
            ],
            "cmpHash" : "default",
            "class" : "VLAN",
            "tag" : 4092,
            "mtu" : 1450
         },
         "net_tap661eb5ae_89" : {
            "interfaces" : [
               {
                  "tagged" : false,
                  "name" : "1.1"
               }
            ],
            "cmpHash" : "default",
            "class" : "VLAN",
            "tag" : 4093,
            "mtu" : 1450
         }
      },
      "label" : "generated from OpenStack metadata",
      "class" : "Device"
   },
   "id" : "d3ab0e60-b9da-46e1-9ccd-a9782cda1481",
   "code" : 422,
   "result" : {
      "errors" : [
         "macAddress is required when assignmentType is UNREACHABLE",
         "macAddress is required when assignmentType is UNREACHABLE"
      ],
      "status" : "ERROR",
      "class" : "Result",
      "message" : "invalid config - rolled back",
      "code" : 422
   }
}
```
restnoded.log

```
Sun, 09 Feb 2020 16:45:52 GMT - fine: [f5-declarative-onboarding: restWorker.js] Getting license from provider.
Sun, 09 Feb 2020 16:45:52 GMT - fine: [f5-declarative-onboarding: restWorker.js] Licensing from pool BIGIPVEREGKEYS
Sun, 09 Feb 2020 16:45:52 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/identified-devices/config/device-info
Sun, 09 Feb 2020 16:45:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"baseMac":"FA:16:3E:1B:8F:C5","halUuid":"4f7ef43b-da12-244f-bf25-98c493fc50a6","chassisSerialNumber":"4f7ef43b-da12-244f-98c493fc50a6","license":{"generation":0,"lastUpdateMicros":1581266002356105},"interfaces":[],"isIControlRestSupported":true,"icrdPort":8100,"time":1581266753609,"physicalMemory":6144,"platform":"Z100","cpu":"Intel Core Processor (Skylake, IBRS)","machineId":"d81134db-c0c2-4ffd-9901-a239e3f04312","hostname":"waf4nic-secondary.openstack.grubernet.org","version":"14.1.2","product":"BIG-IP","platformMarketingName":"BIG-IP Virtual Edition","edition":"Final","build":"0.0.37","restFrameworkVersion":"14.1.2-0.0.37","managementAddress":"192.168.245.37","mcpDeviceName":"/Common/bigip1","isClustered":false,"isVirtual":true,"generation":0,"lastUpdateMicros":0,"kind":"shared:resolver:device-groups:deviceinfostate","selfLink":"https://localhost/mgmt/shared/identified-devices/config/device-info"}
Sun, 09 Feb 2020 16:45:53 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: got error {"code":400}
Sun, 09 Feb 2020 16:45:53 GMT - finest: [f5-declarative-onboarding: restWorker.js] typeof err object
Sun, 09 Feb 2020 16:45:53 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: macAddress is required when assignmentType is UNREACHABLE tries left: 30
```

### Expected Behavior
DO resolves the baseMAC and gets a license from BIG-IQ

### Actual Behavior
BIG-IQ errors says request is missing macAddress attribute

**Comment by jgruber** [comment]
Same License declaration on two BIG-IPs.. one works one does not.

The one that does not is has a different 'device-info' output than the one that does.. Notice the 'hostMac' is not guaranteed to be populated. 'baseMac' should be.

Look what line 182 of f5-cloud-libs bigIq54LicenseProvider.js is using to populate the license request:
```
macAddress: deviceInfo.hostMac
```
No check if that attribute is even populated. Just assumed it would be. Why are we not using the baseMac of the device for VE?

On a BIG-IP which is failing:
```
[root@waf4nic-secondary:NO LICENSE:Standalone] config # curl -s -u 'admin:' http://localhost:8100/shared/identified-devices/config/device-info|json_pp
{
   "machineId" : "d81134db-c0c2-4ffd-9901-a239e3f04312",
   "restFrameworkVersion" : "14.1.2-0.0.37",
   "physicalMemory" : 6144,
   "time" : 1581268344904,
   "isClustered" : false,
   "platformMarketingName" : "BIG-IP Virtual Edition",
   "hostname" : "waf4nic-secondary.openstack.grubernet.org",
   "selfLink" : "https://localhost/mgmt/shared/identified-devices/config/device-info",
   "kind" : "shared:resolver:device-groups:deviceinfostate",
   "interfaces" : [],
   "edition" : "Final",
   "mcpDeviceName" : "/Common/bigip1",
   "platform" : "Z100",
   "cpu" : "Intel Core Processor (Skylake, IBRS)",
   "icrdPort" : 8100,
   "version" : "14.1.2",
   "lastUpdateMicros" : 0,
   "build" : "0.0.37",
   "license" : {
      "generation" : 0,
      "lastUpdateMicros" : "1581266002356105"
   },
   "isVirtual" : true,
   "generation" : 0,
   "halUuid" : "4f7ef43b-da12-244f-bf25-98c493fc50a6",
   "managementAddress" : "192.168.245.37",
   "product" : "BIG-IP",
   "isIControlRestSupported" : true,
   "chassisSerialNumber" : "4f7ef43b-da12-244f-98c493fc50a6",
   "baseMac" : "FA:16:3E:1B:8F:C5"
}
```
On a BIG-IP that is working:
```
[root@waf4nic-primary:Active:Standalone] config # curl -s -u 'admin:' http://localhost:8100/shared/identified-devices/config/device-info|json_pp
{
   "machineId" : "aeb5ee10-1a12-4e62-9e51-833d5059cf8f",
   "restFrameworkVersion" : "14.1.2-0.0.37",
   "physicalMemory" : 6144,
   "time" : 1581268437605,
   "isClustered" : false,
   "platformMarketingName" : "BIG-IP Virtual Edition",
   "hostname" : "waf4nic-primary.openstack.grubernet.org",
   "selfLink" : "https://localhost/mgmt/shared/identified-devices/config/device-info",
   "kind" : "shared:resolver:device-groups:deviceinfostate",
   "interfaces" : [
      "1.1",
      "1.2",
      "1.3",
      "mgmt"
   ],
   "edition" : "Final",
   "mcpDeviceName" : "/Common/waf4nic-primary.openstack.grubernet.org",
   "address" : "1.1.1.25",
   "platform" : "Z100",
   "cpu" : "Intel Core Processor (Skylake, IBRS)",
   "version" : "14.1.2",
   "icrdPort" : 8100,
   "lastUpdateMicros" : 0,
   "build" : "0.0.37",
   "license" : {
      "generation" : 0,
      "activeModules" : [
         "BIG-IP, LAB(LTM,AFM,APM,ASM,AM, DNS, PSM,GTM),VE|ZJPJLZE-HWDGPEM|IPV6 Gateway|Rate Shaping|External Interface and Network HSM, VE|SDN Services, VE|SSL, Forward Proxy, VE|WOM, VE|DNS Services (LAB)|SSL, VE|WBA, VE|BIG-IP VE, Multicast Routing|APM, Limited|Routing Bundle, VE|ASM, VE|Acceleration Manager, VE|Max Compression, VE|Ram Cache|Advanced Protocols, VE|AFM, VE (LAB ONLY - NO ROUTING)|Anti-Virus Checks|Base Endpoint Security Checks|Firewall Checks|Network Access|Secure Virtual Keyboard|APM, Web Application|Machine Certificate Checks|Protected Workspace|Remote Desktop|App Tunnel|AAM, Upgrade from WAM, (v11.4 & later)|VE, Carrier Grade NAT (AFM ONLY)|PSM, VE|DNS Rate Fallback, Unlimited|DNS Licensed Objects, Unlimited|DNS Rate Limit, Unlimited QPS|BIG-IP, LAB(LTM,AFM,APM,ASM,AM, DNS, PSM,GTM),VE|DNSSEC",
         "BIG-IP, LAB(LTM,AFM,APM,ASM,AM, DNS, PSM,GTM),VE|ZJPJLZE-HWDGPEM|IPV6 Gateway|Rate Shaping|External Interface and Network HSM, VE|SDN Services, VE|SSL, Forward Proxy, VE|WOM, VE|DNS Services (LAB)|SSL, VE|WBA, VE|BIG-IP VE, Multicast Routing|APM, Limited|Routing Bundle, VE|ASM, VE|Acceleration Manager, VE|Max Compression, VE|Ram Cache|Advanced Protocols, VE|AFM, VE (LAB ONLY - NO ROUTING)|DNSSEC|Anti-Virus Checks|Base Endpoint Security Checks|Firewall Checks|Network Access|Secure Virtual Keyboard|APM, Web Application|Machine Certificate Checks|Protected Workspace|Remote Desktop|App Tunnel|AAM, Upgrade from WAM, (v11.4 & later)|VE, Carrier Grade NAT (AFM ONLY)|PSM, VE|DNS Rate Fallback, Unlimited|DNS Licensed Objects, Unlimited|DNS Rate Limit, Unlimited QPS|BIG-IP, LAB(LTM,AFM,APM,ASM,AM, DNS, PSM,GTM),VE",
         "CGN, VE (LAB)|CCASFFW-TLEONOU|Routing Bundle, VE|CGN, VE (LAB)",
         "DNSSEC|YAULOXW-SMRHOPQ|DNSSEC"
      ],
      "licenseEndDateTime" : "2020-03-25T00:00:00Z",
      "lastUpdateMicros" : "1581266336440421",
      "registrationKey" : "KJYZS-SYURE-QMUBS-MWGKW-INLDVON"
   },
   "isVirtual" : true,
   "generation" : 0,
   "halUuid" : "d8b2f3d7-34a5-294a-9820-5f06ec757d22",
   "hostMac" : "FA:16:3E:5B:95:C2",
   "managementAddress" : "192.168.245.27",
   "product" : "BIG-IP",
   "isIControlRestSupported" : true,
   "slots" : [
      {
         "volume" : "HD1.1",
         "isActive" : true,
         "version" : "14.1.2",
         "product" : "BIG-IP",
         "build" : "0.0.37"
      }
   ],
   "chassisSerialNumber" : "d8b2f3d7-34a5-294a-5f06ec757d22",
   "baseMac" : "FA:16:3E:5B:95:C2"
}
```

**Comment by jgruber** [comment]
Need to fixes:

- DO schema should allow for the explicit declaration of the 'macAddress' on the License class, allowing flexibility in this value
- cloud-libs should be resolving on device-info 'baseMac' for VE.

**Comment by dstokesf5** [comment]
We got the suggested modification into cloud-libs, and the fix should be available in the next DO release.

**Comment by dstokesf5** [comment]
This was resolved in the 1.11.0 release.
